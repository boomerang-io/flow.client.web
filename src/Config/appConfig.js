// Look for the data injected into the HTML file from the Express app
// See server/app.js for implementation
export const APP_ROOT =
  window._SERVER_DATA && window._SERVER_DATA.APP_ROOT ? window._SERVER_DATA.APP_ROOT : "/BMRG_APP_ROOT_CONTEXT";

export const PRODUCT_STANDALONE =
  window._SERVER_DATA && window._SERVER_DATA.PRODUCT_STANDALONE
    ? window._SERVER_DATA.PRODUCT_STANDALONE === "true"
    : process.env.REACT_APP_PRODUCT_STANDALONE
    ? true
    : false;

export const EMBEDDED_MODE =
  window._SERVER_DATA && window._SERVER_DATA.EMBEDDED_MODE
    ? window._SERVER_DATA.EMBEDDED_MODE === "true"
    : process.env.REACT_APP_EMBEDDED_MODE
    ? true
    : true;

export const CORE_ENV_URL =
  window._SERVER_DATA && window._SERVER_DATA.CORE_ENV_URL ? window._SERVER_DATA.CORE_ENV_URL : "";

export const BASE_DOCUMENTATION_URL = PRODUCT_STANDALONE
  ? "https://www.useboomerang.io/docs/boomerang-flow"
  : `${CORE_ENV_URL}/docs/boomerang-flow`;

export const isDevEnv = process.env.NODE_ENV === "development";
export const isTestEnv = process.env.NODE_ENV === "test";

export const AppPath = {
  Root: "/",
  Error: "/error",
  Activity: "/activity",
  Editor: "/editor/:workflowId",
  EditorDesigner: `/editor/:workflowId/workflow`,
  EditorConfigure: `editor/:workflowId/configure`,
  EditorChangelog: `/editor/:workflowId/changelog`,
  EditorProperties: `/editor/:workflowId/parameters`,
  Execution: "/activity/:workflowId/execution/:executionId",
  Insights: "/insights",
  Properties: "/admin/parameters",
  Quotas: "/admin/quotas",
  QuotasEdit: "/admin/quotas/:teamId",
  Settings: "/admin/settings",
  SystemWorkflows: "/admin/system-workflows",
  TaskTemplates: "/admin/task-templates",
  TaskTemplateEdit: `/admin/task-templates/:id/:version`,
  TaskTemplateYaml: `/admin/task-templates/:id/:version/yaml-editor`,
  Team: "/admin/teams/:teamId",
  TeamSettings: "/admin/teams/:teamId/settings",
  TeamWorkflows: "/admin/teams/:teamId/workflows",
  TeamList: "/admin/teams",
  TeamProperties: `/manage/team-parameters`,
  User: "/admin/users/:userId",
  UserList: "/admin/users",

  Workflows: "/workflows",
  WorkflowsMine: "/workflows/mine",
  WorkflowsTeams: "/workflows/teams",

  ManageTaskTemplates: `/manage/task-templates`,
  ManageTaskTemplatesTeam: `/manage/task-templates/team/:teamId`,
  ManageTaskTemplateEdit: `/manage/task-templates/team/:teamId/:taskId/:version`,
  ManageTaskTemplateYaml: `/manage/task-templates/team/:teamId/:taskId/:version/yaml-editor`,
};

export const appLink = {
  activity: () => `/activity`,
  editorDesigner: ({ workflowId }) => `/editor/${workflowId}/workflow`,
  editorConfigure: ({ workflowId }) => `/editor/${workflowId}/configure`,
  editorChangelog: ({ workflowId }) => `/editor/${workflowId}/changelog`,
  editorProperties: ({ workflowId }) => `/editor/${workflowId}/parameters`,
  execution: ({ executionId, workflowId }) => `/activity/${workflowId}/execution/${executionId}`,
  insights: () => "/insights",
  properties: () => "/admin/parameters",
  manageUsers: () => "/admin/users",
  quotas: () => "/admin/quotas",
  quotasEdit: ({ teamId }) => `/admin/quotas/${teamId}`,
  settings: () => "/admin/settings",
  systemWorkflows: () => "/admin/system-workflows",
  taskTemplates: () => "/admin/task-templates",
  taskTemplateEdit: ({ id, version }) => `/admin/task-templates/${id}/${version}`,
  taskTemplateYaml: ({ id, version }) => `/admin/task-templates/${id}/${version}/yaml-editor`,
  teamProperties: () => `/manage/team-parameters`,
  teamTaskTemplates: () => `/manage/task-templates`,
  team: ({ teamId }) => `/admin/teams/${teamId}`,
  teamWorkflows: ({ teamId }) => `/admin/teams/${teamId}/workflows`,
  teamSettings: ({ teamId }) => `/admin/teams/${teamId}/settings`,
  teamList: () => "/admin/teams",
  workflows: () => "/workflows",
  workflowsMine: () => "/workflows/mine",
  workflowsTeams: () => "/workflows/teams",
  workflowActivity: ({ workflowId }) => `/activity?page=0&size=10&workflowIds=${workflowId}`,

  manageTaskTemplates: ({ teamId }) => `/manage/task-templates/team/${teamId}`,
  manageTaskTemplateEdit: ({ teamId, taskId, version }) => `/manage/task-templates/team/${teamId}/${taskId}/${version}`,
  manageTaskTemplateYaml: ({ teamId, taskId, version }) =>
    `/manage/task-templates/team/${teamId}/${taskId}/${version}/yaml-editor`,

  //external apps
  docsWorkflowEditor: () => `${BASE_DOCUMENTATION_URL}/how-to-guide/workflow-editor`,
};

export const queryStringOptions = { arrayFormat: "comma", skipEmptyString: true };

export const FeatureFlag = {
  /**
   * new Feature Flags
   */
  TeamManagementEnabled: "TeamManagementEnabled",
  WorkflowQuotasEnabled: "WorkflowQuotasEnabled",
  SettingsEnabled: "SettingsEnabled",
  UserManagementEnabled: "UserManagementEnabled",
  GlobalParametersEnabled: "GlobalParametersEnabled",
  WorkflowTokensEnabled: "WorkflowTokensEnabled",
  TaskManagerEnabled: "TaskManagerEnabled",
  EditVerifiedTasksEnabled: "EditVerifiedTasksEnabled",
  WorkflowTriggersEnabled: "WorkflowTriggersEnabled",
  TeamParametersEnabled: "TeamParametersEnabled",

  ActivityEnabled: "ActivityEnabled",
  InsightsEnabled: "InsightsEnabled",
};
