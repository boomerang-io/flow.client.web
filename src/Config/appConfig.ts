// Look for the data injected into the HTML file from the Express app
// See server/app.js for implementation
import { Envs } from "Constants";
import { StringifyOptions } from "query-string";
export const APP_ROOT =
  window._SERVER_DATA && window._SERVER_DATA.APP_ROOT ? window._SERVER_DATA.APP_ROOT : "/BMRG_APP_ROOT_CONTEXT";

export const CORE_ENV_URL =
  window._SERVER_DATA && window._SERVER_DATA.CORE_ENV_URL ? window._SERVER_DATA.CORE_ENV_URL : "";

export const BASE_DOCUMENTATION_URL = "https://www.useboomerang.io/docs/boomerang-flow";

//@ts-ignore
export const isDevEnv = import.meta.env.MODE === Envs.Dev;
//@ts-ignore
export const isTestEnv = import.meta.env.MODE === Envs.Test;

type AppPathKey =
  | "Root"
  | "Error"
  | "Activity"
  | "Actions"
  | "ActionsApprovals"
  | "ActionsManual"
  | "Editor"
  | "EditorDesigner"
  | "EditorConfigure"
  | "EditorChangelog"
  | "EditorProperties"
  | "EditorSchedule"
  | "Execution"
  | "Home"
  | "Profile"
  | "Insights"
  | "ManageTaskTemplates"
  | "ManageTaskTemplateDetail"
  | "ManageTaskTemplateEditor"
  | "ManageTeam"
  | "Properties"
  | "Schedules"
  | "Settings"
  | "TemplateWorkflows"
  | "TaskTemplates"
  | "TaskTemplateDetail"
  | "TaskTemplateEditor"
  | "ManageTeam"
  | "ManageTeamSettings"
  | "ManageTeamWorkflows"
  | "ManageTeamLabels"
  | "ManageTeamQuotas"
  | "ManageTeamApprovers"
  | "ManageTeamProperties"
  | "ManageTeamTokens"
  | "Tokens"
  | "TeamList"
  | "User"
  | "UserList"
  | "UserLabels"
  | "UserSettings"
  | "Workflows";

export const AppPath: Record<AppPathKey, string> = {
  Root: "/",
  Error: "/error",
  Activity: "/:teamId/activity",
  Actions: "/:teamId/actions",
  ActionsApprovals: "/:teamId/actions/approvals",
  ActionsManual: "/:teamId/actions/manual",
  Editor: "/:teamId/editor/:workflowId",
  EditorDesigner: `/:teamId/editor/:workflowId/workflow`,
  EditorConfigure: `/:teamId/editor/:workflowId/configure`,
  EditorChangelog: `/:teamId/editor/:workflowId/changelog`,
  EditorProperties: `/:teamId/editor/:workflowId/parameters`,
  EditorSchedule: `/:teamId/editor/:workflowId/schedule`,
  Execution: "/:teamId/activity/:workflowId/execution/:executionId",
  Home: "/home",
  Profile: "/profile",
  Insights: "/:teamId/insights",
  Workflows: "/:teamId/workflows",
  Schedules: "/:teamId/schedules",

  //Manage
  ManageTaskTemplates: `/:teamId/task-templates`,
  ManageTaskTemplateDetail: `/:teamId/task-templates/:name/:version`,
  ManageTaskTemplateEditor: `/:teamId/task-templates/:name/:version/editor`,
  ManageTeamTokens: "/:teamId/tokens",
  ManageTeamProperties: `/:teamId/parameters`,
  ManageTeam: `/:teamId/manage`,
  ManageTeamSettings: "/:teamId/manage/settings",
  ManageTeamWorkflows: "/:teamId/manage/workflows",
  ManageTeamQuotas: "/:teamId/manage/quotas",
  ManageTeamLabels: "/:teamId/manage/Labels",
  ManageTeamApprovers: `/:teamId/manage/approver-groups`,

  //Admin
  Properties: "/admin/parameters",
  Settings: "/admin/settings",
  TemplateWorkflows: "/admin/template-workflows",
  TaskTemplates: "/admin/task-templates",
  TaskTemplateDetail: `/admin/task-templates/:name/:version`,
  TaskTemplateEditor: `/admin/task-templates/:name/:version/editor`,
  TeamList: "/admin/teams",
  Tokens: "/admin/tokens",
  User: "/admin/users/:userId",
  UserLabels: "/admin/users/:userId/labels",
  UserSettings: "/admin/users/:userId/settings",
  UserList: "/admin/users",
};

interface WorkflowIdArg {
  workflowId: string;
}
interface TeamIdArg {
  teamId: string;
}

interface UserIdArg {
  userId: string
}

type TeamRouteArgs = WorkflowIdArg & TeamIdArg;
interface ManageTaskTemplateArgs {
  teamId: string;
  name: string;
  version: string;
}
interface AdminTaskTemplateArgs {
  name: string;
  version: string;
}
interface ExecutionArgs {
  executionId: string;
  workflowId: string;
}

export const appLink = {
  activity: ({ teamId }: TeamIdArg) => `/${teamId}/activity`,
  actions: ({ teamId }: TeamIdArg) => `/${teamId}/actions`,
  actionsApprovals: ({ teamId }: TeamIdArg) => `/${teamId}/actions/approvals`,
  actionsManual: ({ teamId }: TeamIdArg) => `/${teamId}/actions/manual`,
  editorDesigner: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/editor/${workflowId}/workflow`,
  editorConfigure: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/editor/${workflowId}/configure`,
  editorChangelog: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/editor/${workflowId}/changelog`,
  editorProperties: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/editor/${workflowId}/parameters`,
  editorSchedule: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/editor/${workflowId}/schedule`,
  execution: ({ executionId, workflowId }: ExecutionArgs) => `/activity/${workflowId}/execution/${executionId}`,
  home: () => "/home",
  profile: () => "/profile",
  insights: () => "/insights",
  manageTaskTemplates: ({ teamId }: TeamIdArg) =>
    `/${teamId}/task-templates`,
  manageTaskTemplateEdit: ({ teamId, name, version }: ManageTaskTemplateArgs) =>
    `/${teamId}/task-templates/${name}/${version}`,
  manageTaskTemplateYaml: ({ teamId, name, version }: ManageTaskTemplateArgs) =>
    `/${teamId}/task-templates/${name}/${version}/editor`,
  manageTeam: ({ teamId }: TeamIdArg) => `/${teamId}/manage`,
  manageTeamApprovers: ({ teamId }: TeamIdArg) => `/${teamId}/manage/approver-groups`,
  manageTeamWorkflows: ({ teamId }: TeamIdArg) => `/${teamId}/manage/workflows`,
  manageTeamLabels: ({ teamId }: TeamIdArg) => `/${teamId}/manage/labels`,
  manageTeamQuotas: ({ teamId }: TeamIdArg) => `/${teamId}/manage/quotas`,
  manageTeamSettings: ({ teamId }: TeamIdArg) => `/${teamId}/manage/settings`,
  manageTeamTokens: ({ teamId }: TeamIdArg) => `/${teamId}/tokens`,
  manageTeamParameters: ({ teamId }: TeamIdArg) => `/${teamId}/parameters`,
  manageUsers: () => "/admin/users",
  properties: () => "/admin/parameters",
  schedule: () => "/schedule",
  settings: () => "/admin/settings",
  templateWorkflows: () => "/admin/template-workflows",
  taskTemplates: () => "/admin/task-templates",
  taskTemplateDetail: ({ name, version }: AdminTaskTemplateArgs) => `/admin/task-templates/${name}/${version}`,
  taskTemplateEditor: ({ name, version }: AdminTaskTemplateArgs) => `/admin/task-templates/${name}/${version}/editor`,
  teamList: () => "/admin/teams",
  tokens: () => `/admin/tokens`,
  user: ({ userId }: UserIdArg) => `/admin/users/${userId}`,
  userLabels: ({ userId }: UserIdArg) => `/admin/users/${userId}/labels`,
  userSettings: ({ userId }: UserIdArg) => `/admin/users/${userId}/settings`,
  userList: () => "/admin/users",
  workflows: ({ teamId }: TeamIdArg) => `/${teamId}/workflows`,
  workflowActivity: ({ teamId, workflowId }: TeamRouteArgs) => `/${teamId}/activity?page=0&size=10&workflows=${workflowId}`,
  //external apps
  docsWorkflowEditor: () => `${BASE_DOCUMENTATION_URL}/how-to-guide/workflow-editor`,
};

/**
 * new Feature Flags
 */
export enum FeatureFlag {
  ActivityEnabled = "ActivityEnabled",
  EditVerifiedTasksEnabled = "EditVerifiedTasksEnabled",
  GlobalParametersEnabled = "GlobalParametersEnabled",
  InsightsEnabled = "InsightsEnabled",
  TeamManagementEnabled = "TeamManagementEnabled",
  TeamParametersEnabled = "TeamParametersEnabled",
  TeamTasksEnabled = "TeamTasksEnabled",
  UserManagementEnabled = "UserManagementEnabled",
  WorkflowQuotasEnabled = "WorkflowQuotasEnabled",
  WorkflowTokensEnabled = "WorkflowTokensEnabled",
  WorkflowTriggersEnabled = "WorkflowTriggersEnabled",
}

export const queryStringOptions: StringifyOptions = { arrayFormat: "comma", skipEmptyString: true };
