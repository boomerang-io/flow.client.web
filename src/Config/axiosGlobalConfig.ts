// Setup global config for Axios requests so they work with ICP
import axios from "axios";
axios.defaults.withCredentials = true;
