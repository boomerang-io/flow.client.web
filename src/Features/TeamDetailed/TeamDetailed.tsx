import React from "react";
import { Helmet } from "react-helmet";
import { useQuery } from "react-query";
import { useFeature } from "flagged";
import { Switch, Route, useRouteMatch, useHistory } from "react-router-dom";
import { useAppContext } from "Hooks";
import { Box } from "reflexbox";
import {
  ErrorMessage,
  FeatureHeader,
  FeatureHeaderTitle as HeaderTitle,
  FeatureHeaderSubtitle as HeaderSubtitle,
  Loading,
} from "@boomerang-io/carbon-addons-boomerang-react";
import Header from "./Header";
import Members from "./Members";
import Settings from "./Settings";
import Quotas from "./Quotas";
import Labels from "./Labels";
import Tokens from "./Tokens";
import Workflows from "./Workflows";
import ApproverGroups from "./ApproverGroups";
import { AppPath, FeatureFlag } from "Config/appConfig";
import { serviceUrl, resolver } from "Config/servicesConfig";
import styles from "./teamDetailed.module.scss";

const FeatureLayout: React.FC = ({ children }) => {
  return (
    <>
      <Helmet>
        <title>Teams</title>
      </Helmet>
      <FeatureHeader
        includeBorder={false}
        header={
          <>
            <HeaderTitle style={{ margin: "0" }}>Teams</HeaderTitle>
            <HeaderSubtitle>View and manage your teams</HeaderSubtitle>
          </>
        }
      />
      <Box p="1rem">{children}</Box>
    </>
  );
};

function TeamDetailedContainer() {
  const teamManagementEnabled = useFeature(FeatureFlag.TeamManagementEnabled);
  const match: { params: { teamId: string } } = useRouteMatch();
  const teamId = match?.params?.teamId;
  const { user } = useAppContext();

  const teamDetailsUrl = serviceUrl.getTeam({ teamId });

  const {
    data: teamDetailsData,
    error: teamDetailsError,
    isLoading: teamDetailsIsLoading,
  } = useQuery({
    queryKey: teamDetailsUrl,
    queryFn: resolver.query(teamDetailsUrl),
  });

  if (teamDetailsIsLoading)
    return (
      <FeatureLayout>
        <Loading />
      </FeatureLayout>
    );
  if (teamDetailsError)
    return (
      <FeatureLayout>
        <ErrorMessage />
      </FeatureLayout>
    );

  if (teamDetailsData) {
    const canEdit = teamManagementEnabled && teamDetailsData.status === "active";
    // const teamOwnerIdList = teamDetailsData?.owners?.map((owner) => owner.ownerId);
    return (
      <div className={styles.container}>
        <Header team={teamDetailsData} />
        <Switch>
          <Route exact path={AppPath.ManageTeam}>
            <Members
              canEdit={canEdit}
              team={teamDetailsData}
              memberList={teamDetailsData.members}
              user={user}
              teamDetailsUrl={teamDetailsUrl}
            />
          </Route>
          <Route exact path={AppPath.ManageTeamWorkflows}>
            <Workflows team={teamDetailsData} />
          </Route>
          <Route exact path={AppPath.ManageTeamApprovers}>
            <ApproverGroups team={teamDetailsData} canEdit={canEdit} />
          </Route>
          <Route exact path={AppPath.ManageTeamQuotas}>
            <Quotas team={teamDetailsData} canEdit={canEdit && user?.type === "admin"} />
          </Route>
          <Route exact path={AppPath.ManageTeamTokens}>
            <Tokens team={teamDetailsData} />
          </Route>
          <Route exact path={AppPath.ManageTeamLabels}>
            <Labels team={teamDetailsData} canEdit={canEdit} />
          </Route>
          <Route exact path={AppPath.ManageTeamSettings}>
            <Settings team={teamDetailsData} canEdit={canEdit} />
          </Route>
        </Switch>
      </div>
    );
  }

  return null;
}

export default TeamDetailedContainer;
