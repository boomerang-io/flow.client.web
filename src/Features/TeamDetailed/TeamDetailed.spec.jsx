import React from "react";
import TeamDetailed from ".";
import { Route } from "react-router-dom";
import { screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AppPath, appLink } from "Config/appConfig";
import { startApiServer } from "ApiServer";

jest.setTimeout(60000);
let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
});

describe("TeamDetailed --- Snapshot Test", () => {
  it("Capturing Snapshot of TeamDetailed", async () => {
    const { baseElement } = rtlContextRouterRender(
      <Route path={AppPath.Team}>
        <TeamDetailed />
      </Route>,
      { route: appLink.team({ teamId: "5e7cccb94bbc6e0001c51773" }) }
    );
    await screen.findByText("These are the people who have access to workflows for this Team.");
    expect(baseElement).toMatchSnapshot();
  });
});

describe("TeamDetailed --- RTL", () => {
  test("Visit Team Details tabs", async () => {
    rtlContextRouterRender(
      <Route path={AppPath.Team}>
        <TeamDetailed />
      </Route>,
      { route: appLink.team({ teamId: "5e7cccb94bbc6e0001c51773" }) }
    );
    //Members tab
    await screen.findByText("These are the people who have access to workflows for this Team.");
    const addMemberButton = await screen.findByText(/^Add Members$/i);
    fireEvent.click(addMemberButton);
    expect(screen.getByText(/^Search for people to add to this team$/i)).toBeInTheDocument();
    expect(screen.getByText(/^Add to team$/i)).toBeDisabled();

    expect(screen.getByPlaceholderText(/^Search for a user$/i)).toBeInTheDocument();
    // fetch users is triggering a promise error, need to look for a solution
    // const nameInput = screen.getByPlaceholderText(/^Search for a user$/i);
    // userEvent.type(nameInput, "e");
    // fireEvent.click(await screen.findByText(/^Test User$/i));
    // expect(await screen.findByText(/^Add to team$/i)).toBeEnabled();
    fireEvent.click(screen.getByText(/^Cancel$/i));

    //Workflows
    fireEvent.click(screen.getByText("Workflows"));
    expect(screen.getByText("These are the workflows for this Team.")).toBeInTheDocument();

    //Settings
    fireEvent.click(screen.getByText("Settings"));
    expect(await screen.findByText("Team Name")).toBeInTheDocument();
    fireEvent.click(await screen.findByTestId("open-change-name-modal"));
    expect(screen.getByText("Change team name")).toBeInTheDocument();
    userEvent.type(screen.getByLabelText("Name"), " test name");
    fireEvent.click(screen.getByText("Save"));
    expect(await (await screen.findAllByText("WDC2 ISE Dev test name")).length).toBe(3);
  });
});
