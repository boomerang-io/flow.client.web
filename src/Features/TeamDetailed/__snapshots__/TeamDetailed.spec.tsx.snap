// Vitest Snapshot v1

exports[`TeamDetailed --- Snapshot Test > Capturing Snapshot of TeamDetailed 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_1b3db8"
    >
      <header
        class="cds--bmrg-feature-header _container_0b651a --bordered"
      >
        <section
          class="cds--bmrg-feature-header__content"
        >
          <div
            class="cds--bmrg-feature-header__nav"
          >
            <nav
              aria-label="Breadcrumb"
            >
              <ol
                class="cds--breadcrumb cds--breadcrumb--no-trailing-slash"
              >
                <li
                  class="cds--breadcrumb-item"
                >
                  <a
                    class="cds--link"
                    href="/admin/teams"
                  >
                    Teams
                  </a>
                </li>
                <li
                  class="cds--breadcrumb-item cds--breadcrumb-item--current"
                >
                  <p
                    class="cds--link"
                  >
                    WDC2 ISE Dev
                  </p>
                </li>
              </ol>
            </nav>
          </div>
          <div
            class="cds--bmrg-feature-header__header"
          >
            <h1
              class="cds--bmrg-feature-header-title"
            >
              WDC2 ISE Dev
            </h1>
            <div
              class="_subtitle_0b651a"
            >
              <div
                class="_status_0b651a"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  style="fill: #009d9a;"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 24L4 15 5.414 13.586 13 21.171 26.586 7.586 28 9 13 24z"
                  />
                </svg>
                <p
                  class="_statusText_0b651a"
                >
                  Active
                </p>
              </div>
              <span
                class="_statusDivider_0b651a"
              >
                -
              </span>
            </div>
          </div>
          <div
            class="cds--bmrg-feature-header__footer"
          >
            <nav
              aria-label="Team pages"
              class="cds--tabs"
            >
              <ul
                class="cds--tab--list"
              >
                <a
                  aria-current="page"
                  class="cds--tabs__nav-item cds--tabs__nav-link cds--tabs__nav-item--selected"
                  href="/admin/teams/5e7cccb94bbc6e0001c51773"
                >
                  Members
                </a>
                <a
                  class="cds--tabs__nav-item cds--tabs__nav-link"
                  href="/admin/teams/5e7cccb94bbc6e0001c51773/workflows"
                >
                  Workflows
                </a>
                <a
                  class="cds--tabs__nav-item cds--tabs__nav-link"
                  href="/admin/teams/5e7cccb94bbc6e0001c51773/labels"
                >
                  Labels
                </a>
                <a
                  class="cds--tabs__nav-item cds--tabs__nav-link"
                  href="/admin/teams/5e7cccb94bbc6e0001c51773/settings"
                >
                  Settings
                </a>
              </ul>
            </nav>
          </div>
        </section>
        <section
          class="_teamButtons_0b651a"
        >
          <button
            aria-describedby="danger-description-1"
            class="cds--btn cds--btn--md cds--btn--danger"
            data-testid="close-team"
            tabindex="0"
            type="button"
          >
            <span
              class="cds--visually-hidden"
              id="danger-description-1"
            >
              danger
            </span>
            Close Team
            <svg
              aria-hidden="true"
              aria-label="Close"
              class="cds--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
              />
            </svg>
          </button>
        </section>
      </header>
      <section
        aria-label="WDC2 ISE Dev Team Members"
        class="_container_72f9b4"
      >
        <section
          class="_actionsContainer_72f9b4"
        >
          <div
            class="_leftActions_72f9b4"
          >
            <p
              class="_featureDescription_72f9b4"
            >
              These are the people who have access to workflows for this Team.
            </p>
            <p
              class="_memberCountText_72f9b4"
            >
              Showing 
              2
               member
              s
            </p>
            <div
              aria-labelledby="member-search-search"
              class="cds--search cds--search--md"
              role="search"
            >
              <div
                class="cds--search-magnifier"
              >
                <svg
                  aria-hidden="true"
                  class="cds--search-magnifier-icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                  />
                </svg>
              </div>
              <label
                class="cds--label"
                for="member-search"
                id="member-search-search"
              >
                member search
              </label>
              <input
                autocomplete="off"
                class="cds--search-input"
                id="member-search"
                placeholder="Search for a member"
                role="searchbox"
                type="text"
                value=""
              />
              <button
                aria-label="Clear search input"
                class="cds--search-close cds--search-close--hidden"
                title="Clear search input"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="_rightActions_72f9b4"
          >
            <button
              class="cds--btn cds--btn--md cds--btn--primary"
              data-testid="add-members-button"
              tabindex="0"
              type="button"
            >
              Add Members
              <svg
                aria-hidden="true"
                aria-label="Add members"
                class="cds--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                />
              </svg>
            </button>
          </div>
        </section>
        <div
          aria-label="Structured list section"
          class="cds--structured-list"
          role="table"
        >
          <div
            class="cds--structured-list-thead"
            role="rowgroup"
          >
            <div
              class="cds--structured-list-row cds--structured-list-row--header-row"
              role="row"
            >
              <span
                class="cds--structured-list-th"
                role="columnheader"
              >
                Name
              </span>
              <span
                class="cds--structured-list-th"
                role="columnheader"
              >
                Email
              </span>
              <span
                class="cds--structured-list-th"
                role="columnheader"
              >
                Added on
              </span>
              <span
                class="cds--structured-list-th"
                role="columnheader"
              />
              <span
                class="cds--structured-list-th"
                role="columnheader"
              />
              <span
                class="cds--structured-list-th"
                role="columnheader"
              />
            </div>
          </div>
          <div
            class="cds--structured-list-tbody"
            role="rowgroup"
          >
            <div
              class="cds--structured-list-row"
              role="row"
            >
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <div
                  class="_memberNameContainer_72f9b4"
                >
                  <p>
                    Marcus Roy
                  </p>
                </div>
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                mdroy@us.ibm.com
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                July 21, 2020
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <button
                  class="_removeButton_aa92f0"
                >
                  Remove from Team
                  <svg
                    aria-hidden="true"
                    fill="#f94d56"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12H14V24H12zM18 12H20V24H18z"
                    />
                    <path
                      d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <a
                  class="_viewMemberLink_72f9b4"
                  href="/admin/users/kg3p380ad09662744c240e47"
                >
                  View user
                </a>
              </div>
            </div>
            <div
              class="cds--structured-list-row"
              role="row"
            >
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <div
                  class="_memberNameContainer_72f9b4"
                >
                  <p>
                    Tim Bula
                  </p>
                </div>
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                trbula@us.ibm.com
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                July 21, 2020
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <button
                  class="_removeButton_aa92f0"
                >
                  Remove from Team
                  <svg
                    aria-hidden="true"
                    fill="#f94d56"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12H14V24H12zM18 12H20V24H18z"
                    />
                    <path
                      d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="cds--structured-list-td"
                role="cell"
              >
                <a
                  class="_viewMemberLink_72f9b4"
                  href="/admin/users/5f17380ad09662744c240e47"
                >
                  View user
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
</body>
`;
